### PR TITLE
Improve cuTENSOR installer

### PR DESCRIPTION
**Pull request by kmaehashi**
* Use `lib` instead of `lib64` (keep dir name in original archive)
  ~Depends on https://github.com/cupy/cupy/pull/4912~ merged
* Preserve license file
* A bit cleanup

### CODE REVIEW

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by kmaehashi** [comment]
#4912 merged, ready for review.
/test

**Review by emcastillo — APPROVED** [review]

File: cupyx/tools/install_library.py [hunk 1]
```diff
@@ -165,12 +165,17 @@ def install_lib(cuda, prefix, library):
...
         elif library == 'cutensor':
-            include = os.path.join(destination, 'include')
-            lib = os.path.join(destination, 'lib64')
-            shutil.move(os.path.join(tmpdir, 'libcutensor/include'), include)
-            if cuda.startswith('11'):
+            if cuda.startswith('11.'):
                 cuda = '11'
-            shutil.move(os.path.join(tmpdir, 'libcutensor/lib', cuda), lib)
+            shutil.move(
+                os.path.join(tmpdir, 'libcutensor', 'include'),
```
Comment (emcastillo) [inline cupyx/tools/install_library.py:171]: Supper stupid and irrelevant.
Is there any strong reason not to use `Path` https://docs.python.org/3/library/pathlib.html
now that we only support python 3? I feel it is more readable than `os.path.join`.
But we don't need to change it now!

Comment (asi1024) [reply cupyx/tools/install_library.py:171]: Don't need to change it now, but I prefer `pathlib` too.

Comment (kmaehashi) [reply cupyx/tools/install_library.py:171]: Thanks for the suggestion! Agree with that, I'll rewrite it pathlib after merging this one and #4848.

**Review by asi1024 — APPROVED** [review]

**Pull request merged**